src/scripts/main.js

```diff
@@ -1,11 +1,21 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}`,
+      );
+    });
+  });
```

Comment:
The `waitFor` function is correctly implemented to return a promise that resolves when the specified event occurs on the element. However, consider handling potential errors or exceptions that might occur during event handling.